pkg/prometheus/operator.go

```diff
@@ -2034,7 +2034,7 @@ func (c *Operator) selectProbes(ctx context.Context, p *monitoringv1.Prometheus,
 			)
 		}
 		if probe.Spec.Targets.StaticConfig == nil && probe.Spec.Targets.Ingress == nil {
```

Comment:
I believe the panic is introduced when `probe.Spec.Targets` is nil. Can we make sure we handle that case as well?

Reply:
It uses struct instead of a pointer, it would not be `nil`. :D

https://github.com/prometheus-operator/prometheus-operator/blob/dea9a44e53e41ae2ef92d72d7f44eaa6dbe31f32/pkg/apis/monitoring/v1/types.go#L1084

Reply:
Ah, you're right. We return a nil error in this case! Would you mind just adding a unit test to go with the bugfix?

Reply:
> Ah, you're right. We return a nil error in this case! Would you mind just adding a unit test to go with the bugfix?

It's hard to add unit tests for `selectProbes()` I think I would extract these lines as a function and make unit test for it.

Reply:
Updated! PTAL! :heart:

Reply:
Looks great :+1: I would just move the validation function to the probe itself, similar to how we do it for the other types: https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/types.go#L1194

Wdyt?

Reply:
> Looks great +1 I would just move the validation function to the probe itself, similar to how we do it for the other types: [`main`/pkg/apis/monitoring/v1/types.go#L1194](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/types.go?rgh-link-date=2022-01-31T11%3A12%3A31Z#L1194)
> 
> Wdyt?

Updated! PTAL! :D